examples/Getting Started with Semiconductor Module/Getting Started with Semiconductor Module.py

```diff
@@ -0,0 +1,215 @@
+        dc_max_leakage_measurements.append(measurements)
+
+    digital_pin_query.publish(dc_max_leakage_measurements, "DC.vcc_max")
+
+    dc_gnd = tsm_context.get_specifications_value("DC.gnd")
+
+    # Turn off Vcc to 0V and measure leakage current
+    for session, channel_string in zip(dcpower_sessions, dcpower_channel_strings):
+        simulate_set_power_supply_voltage(session, channel_string, dc_gnd)
+
+    dc_ground_leakage_measurements = []
+    for session, pin_set_string in zip(digital_sessions, digital_pin_set_strings):
+        # simulate leakage current
+        measurements = []
+        for _ in pin_set_string.split(","):
+            measurements += simulate_parametric_measurement(1.05e-7, 2.2e-8, 0.19)
+        dc_ground_leakage_measurements.append(measurements)
+
+    digital_pin_query.publish(dc_ground_leakage_measurements, "DC.gnd")
+
+
+def simulate_set_power_supply_voltage(
+    session: nidcpower.Session, channel_string: str, set_voltage: float
+):
+    """Simulate configuring a voltage on an NI-DCPower instrument."""
+
+
+@code_module
+def functional(
+    tsm_context: SemiconductorModuleContext,
+    pins,
+):
+    """Test DUT functionality using Digital patterns.
+
+    This example uses simulated data and driver calls so it can run on a system without the
+    NI-Digital Pattern driver installed. Refer to the Accelerometer example for an example
+    of a functional test using the driver.
+    """
+    filtered_pins = tsm_context.filter_pins_by_instrument_type(
+        pins, InstrumentTypeIdConstants.NI_DIGITAL_PATTERN, Capability.ALL
+    )
+    pin_query, sessions, site_lists = tsm_context.pins_to_nidigital_sessions_for_pattern(
+        filtered_pins
+    )
+
+    # Publish pattern result is a list of dict where each element corresponds to a session
+    pattern_results = []
+    for session, site_list in zip(sessions, site_lists):
+        # simulate site pass/fail data as dict with int siteNumber as key
+        pattern_status = {}
+        for site in site_list.split(","):
+            # remove the text site and retain just the number
+            site = site.replace("site", "")
+            simulated_data = random.random()
+            # Simulates a functional test that bursts a pattern and gets each site failed status.
+            pattern_status[int(site)] = simulated_data > 0.02
+
+        pattern_results.append(pattern_status)
+
+    pin_query.publish_pattern_results(pattern_results)
+
+
+@code_module
+def close_all_instruments_sessions(tsm_context: SemiconductorModuleContext):
+    """Close all instrument sessions."""
+    sessions = tsm_context.get_all_nidcpower_sessions()
+    for session in sessions:
+        session.close()
+
+    sessions = tsm_context.get_all_nidigital_sessions()
+    for session in sessions:
+        session.close()
+
+
+def simulate_parametric_measurement(upper_limit: float, lower_limit: float, std_dev: float):
```

Comment:
any reason this function needs these arguments in the example? the LV code uses constants.

Reply:
I see why now. Can we keep this but make it a subroutine of some more general functions like 
`simulate_leakage_data`?